hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/NumberOfWorkflowsToDeleteMetadataMetricActor.scala

```diff
@@ -0,0 +1,81 @@
+package cromwell.services.metadata.hybridcarbonite
+
+import java.time.OffsetDateTime
+
+import akka.actor.{ActorRef, LoggingFSM, Props, Status}
+import akka.pattern.pipe
+import cats.data.NonEmptyList
+import cromwell.core.instrumentation.InstrumentationPrefixes
+import cromwell.services.MetadataServicesStore
+import cromwell.services.instrumentation.CromwellInstrumentation
+import cromwell.services.metadata.MetadataArchiveStatus
+import cromwell.services.metadata.MetadataArchiveStatus.Archived
+import cromwell.services.metadata.hybridcarbonite.NumberOfWorkflowsToDeleteMetadataMetricActor._
+import cromwell.services.metadata.impl.{MetadataDatabaseAccess, MetadataServiceActor}
+
+class NumberOfWorkflowsToDeleteMetadataMetricActor(override val serviceRegistryActor: ActorRef)
+  extends LoggingFSM[MetadataDeletionMetricsActorState, Unit]
+  with MetadataDatabaseAccess
+  with MetadataServicesStore
+  with CromwellInstrumentation {
+
+  implicit val ec = context.dispatcher
+
+  startWith(WaitingForMetricCalculationRequestOrMetricValue, ())
+
+  when(WaitingForMetricCalculationRequestOrMetricValue) {
+    case Event(NumberOfWorkflowsToDeleteMetadataMetricValue(value), _) =>
+      sendGauge(workflowsToDeleteMetadataMetricPath, value, InstrumentationPrefixes.ServicesPrefix)
+      stay()
+    case Event(CalculateNumberOfWorkflowsToDeleteMetadataMetricValue(currentTimestampMinusDelay), _) =>
+      countRootWorkflowSummaryEntriesByArchiveStatusAndOlderThanTimestamp(
+        MetadataArchiveStatus.toDatabaseValue(Archived),
+        currentTimestampMinusDelay
+      )
+        .map(intVal => FinishedNumberOfWorkflowsToDeleteMetadataMetricValueCalculation(intVal.toLong))
+        .pipeTo(self)
+      goto(MetricCalculationInProgress)
+  }
+
+  when(MetricCalculationInProgress) {
+    case Event(CalculateNumberOfWorkflowsToDeleteMetadataMetricValue, _) =>
+      // metric actor is already busy calculating metric value, so we dismiss this request
+      stay()
+    case Event(NumberOfWorkflowsToDeleteMetadataMetricValue(value), _) =>
+      // pass through the pre-calculated metric value and stay, continuing to wait for DB query to finish
+      sendGauge(workflowsToDeleteMetadataMetricPath, value, InstrumentationPrefixes.ServicesPrefix)
+      stay()
```

Comment:
TOL Is this message expected in this state? If this really should be the same code in both states could it be DRYed out to `whenUnhandled` (or is the comment here important enough to keep the duplication)?

Reply:
Probably not "expected" but certainly possible. We should do something sensible if and when it happens.

Reply:
DRYed